spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpoint.java

```diff
@@ -111,23 +104,17 @@ public void setExchangeName(String exchangeName) {
 		this.exchangeName = exchangeName;
 	}
 
-	/**
-	 * @deprecated in favor of {@link #setExpressionExchangeName}. Will be changed in a future release
-	 * to use an {@link Expression} parameter.
-	 * @param exchangeNameExpression the expression to set.
-	 */
-	@Deprecated
-	public void setExchangeNameExpression(String exchangeNameExpression) {
-		Assert.hasText(exchangeNameExpression);
-		this.exchangeNameExpression = expressionParser.parseExpression(exchangeNameExpression);
+	public void setExchangeNameExpression(Expression exchangeNameExpression) {
```

Comment:
Not sure what you mean regarding "normalize all these things", because that is exactly what I'm doing here with this de-deprecation and setter argument type changing.
The `setExpressionExchangeName()` has been introduced for the parser limitation with the Java Bean nature. That's why we are renaming them now to the more readable name, which was with the existing setter for `String`.

Having all that confuse, I suggest to introduce a new `...ExpressionString(String)` setter do not confuse parser and allow to Java-config users have some convenient during their routine.

With your `socketExpression` introduction I see that you agreed here with me and I can go ahead with the proposed fix not waiting for the 5.0. 4.3 should be good enough for such a refactoring change.
WDYT?